src/components/Attachments/AttachmentCarousel/index.native.tsx

```diff
@@ -17,32 +17,20 @@ import AttachmentCarouselPager from './Pager';
 import type {AttachmentCaraouselOnyxProps, AttachmentCarouselProps} from './types';
 import useCarouselArrows from './useCarouselArrows';
 
-function AttachmentCarousel({
-    report,
-    reportActions,
-    parentReportActions,
-    source,
-    onNavigate = () => {},
-    setDownloadButtonVisibility = () => {},
-    onClose = () => {},
-}: AttachmentCarouselProps) {
+function AttachmentCarousel({report, reportActions, parentReportActions, source, onNavigate, setDownloadButtonVisibility, onClose}: AttachmentCarouselProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const pagerRef = useRef<AttachmentCarouselPagerHandle>(null);
     const [page, setPage] = useState<number>();
     const [attachments, setAttachments] = useState<Attachment[]>([]);
     const {shouldShowArrows, setShouldShowArrows, autoHideArrows, cancelAutoHideArrows} = useCarouselArrows();
-    const [activeSource, setActiveSource] = useState<AttachmentSource | undefined | null>(source);
+    const [activeSource, setActiveSource] = useState<AttachmentSource>(source);
 
     const compareImage = useCallback((attachment: Attachment) => attachment.source === source, [source]);
 
     useEffect(() => {
-        if (!report.parentReportActionID || !parentReportActions || !reportActions) {
-            Navigation.dismissModal();
-            return;
-        }
-        const parentReportAction = parentReportActions[report.parentReportActionID];
-        const attachmentsFromReport = extractAttachmentsFromReport(parentReportAction, reportActions);
+        const parentReportAction = report.parentReportActionID && parentReportActions ? parentReportActions[report.parentReportActionID] : undefined;
+        const attachmentsFromReport = extractAttachmentsFromReport(parentReportAction, reportActions ?? undefined);
```

Comment:
```suggestion
        const attachmentsFromReport = extractAttachmentsFromReport(parentReportAction, reportActions);
```

Update `extractAttachmentsFromReport` to accept types
```js
parentReportAction?: OnyxEntry<ReportAction>, reportActions?: OnyxEntry<ReportActions>
```